sympy/polys/series/ring.py

```diff
@@ -1,109 +1,1048 @@
+            else:
+                return self.from_ground(self.domain.convert(expr))
+
+        series = _rebuild(sympify(expr))
+
+        if prec is None:
+            return series
+        else:
+            R = self.ring
+            s = R(self.to_list(series), prec)
+            return self.from_element(s)
+
+    def from_list(
+        self, lst: list[Er], prec: int | None = None
+    ) -> PowerSeriesElement[Er]:
+        """Create a power series element from a list of coefficients."""
+        R = self.ring
+        s = R.from_list(lst, prec)
+        return self.from_element(s)
+
+    def from_element(self, element: TSeriesElement[Er]) -> PowerSeriesElement[Er]:
+        """Convert a lower power series element to a PowerSeriesElement."""
+        return PowerSeriesElement(self, element)
+
+    def from_int(self, arg: int) -> PowerSeriesElement[Er]:
+        """Convert an integer to a power series element."""
+        g = self.domain_new(arg)
+        return self.from_ground(g)
+
+    def from_ground(self, arg: Er) -> PowerSeriesElement[Er]:
+        """Convert a ground element to a power series element."""
+        R = self.ring
+        s = R.from_list([arg])
+        return self.from_element(s)
+
+    def to_expr(self, element: PowerSeriesElement[Er]) -> Expr:
+        """Convert a power series element to an expression."""
+        dom: Domain[Er] = self.domain
+        coeffs: list[Er] = self.ring.to_list(element.series)
+        prec: int | None = self.ring.series_prec(element.series)
+        result: list[Expr] = []
+        x: Expr = self.symbol
+
+        if coeffs and coeffs[0] != 0:
+            result.append(dom.to_sympy(coeffs[0]))
+
+        for i, coeff in enumerate(coeffs[1:], start=1):
+            if coeff == 0:
+                continue
+            result.append(Mul(dom.to_sympy(coeff), Pow(x, i)))
+
+        if prec is not None:
+            result.append(Order(x**prec))
+
+        return Add(*result)
+
+    def to_list(self, element: PowerSeriesElement[Er]) -> list[Er]:
+        """Returns a list of coefficients of a power series."""
+        return self.ring.to_list(element.series)
+
+    def to_dense(self, element: PowerSeriesElement[Er]) -> dup[Er]:
+        """Returns a dense list coefficients of a power series."""
+        return self.ring.to_dense(element.series)
+
+    def domain_new(self, arg: Er | int) -> Er:
+        """Convert arg to the element of ground domain of ring."""
+        return self.domain.convert(arg, self.domain)
+
+    def ring_new(self, arg: Expr | Er | int) -> PowerSeriesElement[Er]:
+        """Create a power series element from various types."""
+        if isinstance(arg, Expr):
+            return self.from_expr(arg)
+        elif isinstance(arg, int):
+            return self.from_int(arg)
+        elif self.domain.of_type(arg):
+            return self.from_ground(arg)
+        else:
+            return NotImplemented
```

Comment:
This should be `raise TypeError`.

`NotImplemented` should only be returned from binary dunder methods like `__add__`.